src/scenes/UserGalleryPage/QRCode.tsx

```diff
@@ -0,0 +1,207 @@
+        dotsOptionsHelper: {
+          colorType: { single: true, gradient: false },
+          gradient: {
+            linear: true,
+            radial: false,
+            color1: '#6a1a4c',
+            color2: '#6a1a4c',
+            rotation: '0',
+          },
+        },
+        cornersSquareOptions: { type: '', color: '#000000' },
+        cornersSquareOptionsHelper: {
+          colorType: { single: true, gradient: false },
+          gradient: {
+            linear: true,
+            radial: false,
+            color1: '#000000',
+            color2: '#000000',
+            rotation: '0',
+          },
+        },
+        cornersDotOptions: { type: '', color: '#000000' },
+        cornersDotOptionsHelper: {
+          colorType: { single: true, gradient: false },
+          gradient: {
+            linear: true,
+            radial: false,
+            color1: '#000000',
+            color2: '#000000',
+            rotation: '0',
+          },
+        },
+        backgroundOptionsHelper: {
+          colorType: { single: true, gradient: false },
+          gradient: {
+            linear: true,
+            radial: false,
+            color1: '#ffffff',
+            color2: '#ffffff',
+            rotation: '0',
+          },
+        },
+      });
+
+      qrCode.append(ref.current);
+      setQRCodeExists(true);
+    }
+  }, [qrCodeExists, username]);
+
+  return (
+    <>
+      <StyledFullScreenQR showQRCode={showQRCode}>
```

Comment:
is it possible to use the modal component to display the QR code instead of this custom fullscreen component? 
it has some functionality like the close button we might be able to reuse.

also It looks like the full screen QR is always in the DOM even if it's not open, with opacity:0 - ideally we should make it so it's only in the DOM when it's open

Reply:
Hmm, the modal component is styled pretty differently from this fullscreen view. It has a border, is centered in the screen, etc.

We could use the same logic to render a new modal (unstyled, or a new component `FullScreenModal`)

Reply:
@kaitoo1  Handled in https://github.com/gallery-so/gallery/pull/807/commits/03deacd9e9c146361a77f62a2753540386825562